ast/compile.go

```diff
@@ -4017,7 +4083,7 @@ func checkUnusedDeclaredVars(loc *Location, stack *localDeclaredVars, used VarSe
 	return errs
 }
 
-func rewriteSomeDeclStatement(g *localVarGenerator, stack *localDeclaredVars, expr *Expr, errs Errors) (*Expr, Errors) {
+func rewriteSomeDeclStatement(g *localVarGenerator, stack *localDeclaredVars, expr *Expr, errs Errors, strict bool) (*Expr, Errors) {
```

Comment:
I ended up sprinkling these `strict` args in a lot of places ... If they're an annoyance, I could find another solution. Would it be better to attach all these functions to the compiler?

Reply:
I think that's OK, and consistent with the prevalent style of the compiler code... 🤔